### DISCUSSION TITLE
Specification for dialect plugin system in Sequelize 8

### DISCUSSION THREAD

**Discussion by clamx4** [sequelize/sequelize#16982] [Help & Questions] [answered]
Hi,

I am interested in contributing to Sequelize by adding HANA dialect support. I received a recommendation https://github.com/sequelize/sequelize/discussions/16939 to make code changes and merge them after a dialect plugin system is implemented in Sequelize 8. 

I am wondering if there is a specification for the plugin system, particularly the APIs. Understanding the plugin architecture will allow me to design and implement the HANA dialect in a more adaptive way, minimizing the effort required for migration from the Sequelize v7 style implementation to the dialect plugin system.

Additionally, I am curious about the feasibility of adding the HANA dialect to Sequelize 7, similar to the CockroachDb DB https://github.com/sequelize/sequelize/pull/15569. While I understand that adding it to Sequelize 7 is not recommended, possibly due to concerns such as integration tests, I am exploring the possibility of addressing these concerns. If the issues can be resolved, supporting HANA in Sequelize 7 might also be a viable option. Would that be correct?

Thank you for your guidance.

**Comment by dazhuangc** [comment]
Hi,
I hope you're doing well.
I just wanted to follow up on my previous message regarding the Sequelize 8 dialect plugin system and the feasibility of adding the HANA dialect to Sequelize 7. I understand you may be busy, but any insights you could provide would be greatly appreciated.
Thank you for your time and consideration.

**Comment by WikiRik** [comment, marked as answer]
Hi! We have not started work on the dialect plugin system yet, and will not do so before the release of v7.

Since the added experimental support of snowflake and ibmi we have changed our policy for adding new dialects and now require that we can run CI, including integration tests, on the dialects. If that's not feasible for HANA, we recommend forking this repository and maintaining updates from upstream yourself. Otherwise it is indeed possible to add native support for HANA.

**Reply by dazhuangc** [reply]
Thanks!

By the way, could you help me close this discussion as a maintainer? I've switched to this new account and currently don't have the permissions to close it.
